File: atividade-4/026_alter/026_alter.js
// Gerar lista: [1, −1, 2, −2, 3, −3, · · · , n, −n]

const number = 6;

const listGen = (min, max) =>
  [...Array(max + 1).keys()].filter((x) => x > min - 1);

const alter = (n) => {
  const list = listGen(1, n);
  const newList = [];

  list.forEach((x) => {
    newList.push(-x);
    newList.push(x);
  });

  return newList;
};

console.log(alter(number));

File: atividade-4/026_alter/026_alter.hs
module Main where

-- Gerar lista: [1, −1, 2, −2, 3, −3, · · · , n, −n]

alter :: Int -> [Int]
alter 0 = []
alter 1 = [1, -1]
alter n = alter(n - 1) ++ [n] ++ [n * (-1)]

main = do
  a <- readLn :: IO Int
  print $ alter a